src/iuse_actor.cpp

```diff
@@ -2578,6 +2609,17 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
         roll = NEUTRAL;
     }
 
+    if (t->has_quality(quality_sew)) {
```

Comment:
Hardcoding by quality ID like this is very bad practice.
You'll need to express this in terms of actor, not tool quality. And it must not be hardcoded by thread either.